services/src/datasets/storage.rs

```diff
@@ -248,6 +272,42 @@ pub trait DatasetStore: DatasetStorer {
     fn wrap_meta_data(&self, meta: MetaDataDefinition) -> Self::StorageType;
 }
 
+#[typetag::serde(tag = "type")]
+pub trait DatasetProviderDefinition:
+    CloneableDatasetProviderDefinition + Send + Sync + std::fmt::Debug
+{
+    /// create the actual provider for data listing and access
+    fn initialize(self: Box<Self>) -> Result<Box<dyn DatasetProvider>>;
+
+    /// the type of the provider
+    fn type_name(&self) -> String;
+
+    /// name of the external data source
+    fn name(&self) -> String;
```

Comment:
type name is a readable representation of the type, e.g. WCS or ChronicleDB
name of the source is the actual instance of the provider, e.g. DWD Geoserver or Emergencity Backend

type name could be a static &str but name is dynamic

Reply:
Ah, I see. Should we make it static then? Couldn't we have a field directly in the trait? :thinking:

Reply:
it's not possible sadly:
> error[E0038]: the trait `datasets::storage::DatasetProviderDefinition` cannot be made into an object
   --> services/src/datasets/storage.rs:305:16
    |
305 | impl Clone for Box<dyn DatasetProviderDefinition> {
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `datasets::storage::DatasetProviderDefinition` cannot be made into an object
    |
    = help: consider moving `type_name` to another trait
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
   --> services/src/datasets/storage.rs:280:11
    |
276 | pub trait DatasetProviderDefinition:
    |           ------------------------- this trait cannot be made into an object...
...
280 |     const type_name: &'static str;
    |           ^^^^^^^^^ ...because it contains this associated `const`